File: 기타/wdz/2.java
import java.util.ArrayList;
import java.util.List;

class Node {
    public int live;
    public String[] str;

    public Node() {
        this.live = 0;
        this.str = new String[30];
        for(int i=0; i<27; i++){
            str[i] = "";
        }
    }
}

class Solution {
    public String[] solution(String[] code) {
        List<String> ans = new ArrayList<String>();
        int ansIdx = 0;
        Node[] node = new Node[30];
        int prev = -1;
        for(int i=0; i<code.length; i++){
            int now = 0;
            for(int j=0; j<code[i].length(); j++){
                if(code[i].charAt(j)!='.'){
                    if(prev < now){
                        node[now] = new Node();
                    }
                    else if(prev > now){
                        for(int idx=prev; idx>now; idx--){
                            node[idx] = null;
                        }
                    }
                    prev = now;
                    if(code[i].charAt(j) == 'p'){
                        char key = code[i].charAt(j+6);
                        int flag = 0;
                        for(int idx = now; idx>=0; idx--){
                            if(node[idx] == null)   continue;
                            if(node[idx].str[key - 'a'] == "")  continue;
                            else{
                                ans.add(node[idx].str[key - 'a']);
                                flag = 1;
                                break;
                            }
                        }
                        if(flag == 0){
                            ans.add("error");
                        }
                    }
                    else{
                        char key = code[i].charAt(j);
                        String value = code[i].substring(j);
                        node[now].str[key-'a'] = value;
                    }
                    break;
                }
                else    now++;
            }
        }
        String[] answer = new String[ans.size()];
        int size=0;
        for(String temp : ans){
            answer[size++] = temp;
        }
        return answer;
    }
}

File: 기타/HD/2.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<bool> solution(vector<string> a) {
    vector<bool> answer;
    for(int i=0; i<a.size(); i++){
        string str = a[i];
        int cnt = 0;
        for(int j=0; j<str.length(); j++){
            if(str[j] == 'a')   cnt++;
        }
        int l = 0;
        int r = str.length()-1;
        int f = 10;
        while(l!=r){
            if(str[l] == 'b' && str[r] == 'b'){
                int flag = 0;
                if(cnt == 0)    break;
                for(int k=0; k<cnt; k++){
                    if(str[l] == 'b'&&str[r]=='b'){
                        l++;
                        r--;
                    }
                    else{
                        flag = 1;
                        break;
                    }
                }
                if(flag == 1){
                    break;
                }
            }
            else if(str[l] == 'a' && str[r] != 'a'){
                l++;
                cnt--;
            }
            else if(str[l] != 'a' && str[r] == 'a'){
                r--;
                cnt--;
            }
            else if(str[l]=='a' && str[r] == 'a'){
                l++;
                cnt--;
            }
        }
        if(l==r && str[l] == 'a')    answer.push_back(true);
        else        answer.push_back(false);
    }
    return answer;
}